orchestrator-bundle/orc8r-libs/tests/test_orc8r_base_db.py

```diff
@@ -129,43 +123,70 @@ def test_given_ready_when_get_plan_then_plan_is_filled_with_magma_orc8r_dummy_se
 
-    @patch(
-        "charms.magma_orc8r_libs.v0.orc8r_base_db.Orc8rBase._get_db_connection_string",
-        new_callable=PropertyMock,
-    )
-    @patch("charms.magma_orc8r_libs.v0.orc8r_base_db.Orc8rBase._db_relation_created")
-    @patch("charms.magma_orc8r_libs.v0.orc8r_base_db.Orc8rBase._db_relation_ready")
+    @patch("psycopg2.connect", new=Mock())
     def test_db_relation_added_when_get_status_then_status_is_active(
-        self, db_relation_established, db_relation_created, get_db_connection_string
+        self,
     ):
-        db_relation_established.return_value = True
-        db_relation_created.return_value = True
-        get_db_connection_string.return_value = self.TEST_DB_CONNECTION_STRING
+        db_relation_id = self.harness.add_relation(relation_name="db", remote_app="postgresql-k8s")
+        key_values = {"master": self.TEST_DB_CONNECTION_STRING.__str__()}
+        self.harness.update_relation_data(
+            relation_id=db_relation_id, key_values=key_values, app_or_unit="postgresql-k8s"
+        )
 
         self.harness.container_pebble_ready("magma-orc8r-dummy")
 
         self.assertEqual(self.harness.charm.unit.status, ActiveStatus())
 
+    def test_given_db_relation_not_created_when_pebble_ready_then_status_is_blocked(
+        self,
+    ):
+        self.harness.container_pebble_ready(container_name="magma-orc8r-dummy")
+        assert self.harness.charm.unit.status == BlockedStatus(
+            "Waiting for db relation to be created"
+        )
+
+    def test_given_db_relation_not_ready_when_pebble_ready_then_status_is_blocked(self):
+        self.harness.add_relation(relation_name="db", remote_app="postgresql-k8s")
+        self.harness.container_pebble_ready(container_name="magma-orc8r-dummy")
+        assert self.harness.charm.unit.status == WaitingStatus(
+            "Waiting for db relation to be ready"
+        )
+
+    @patch("psycopg2.connect", new=Mock())
     @patch(
         "charms.magma_orc8r_libs.v0.orc8r_base_db.Orc8rBase.namespace",
         PropertyMock(return_value="qwerty"),
     )
+    def test_given_pebble_ready_when_db_relation_broken_then_status_is_blocked(
+        self,
+    ):
+        self.harness.set_can_connect("magma-orc8r-dummy", True)
+        container = self.harness.model.unit.get_container("magma-orc8r-dummy")
+        db_relation_id = self.harness.add_relation(relation_name="db", remote_app="postgresql-k8s")
+        key_values = {"master": self.TEST_DB_CONNECTION_STRING.__str__()}
+        self.harness.update_relation_data(
+            relation_id=db_relation_id, key_values=key_values, app_or_unit="postgresql-k8s"
+        )
+        self.harness.charm.on.magma_orc8r_dummy_pebble_ready.emit(container)
+        assert self.harness.charm.unit.status == ActiveStatus()
+        self.harness.remove_relation(db_relation_id)
+        assert self.harness.charm.unit.status == BlockedStatus(
+            "Waiting for db relation to be created"
+        )
```

Comment:
Small nitpick, but in tests with more than a few lines, I think separating the `given/when/then` sections with a newline helps reading:
```suggestion
        assert self.harness.charm.unit.status == ActiveStatus()
        
        self.harness.remove_relation(db_relation_id)
        
        assert self.harness.charm.unit.status == BlockedStatus(
            "Waiting for db relation to be created"
        )
```